src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -1479,6 +1470,150 @@ static void Main()
         }
 
         [Fact]
+        public void DistinctTupleTypesInCompilation()
+        {
+            var source1 = @"
+public class C1
+{
+    public static (int a, int b) M()
+    {
+        return (1, 2);
+    }
+}
+" + trivial2uple;
+
+            var source2 = @"
+public class C2
+{
+    public static (int c, int d) M()
+    {
+        return (3, 4);
+    }
+}
+" + trivial2uple;
+
+            var source = @"
+class C3
+{
+    public static void Main()
+    {
+        System.Console.Write(C1.M().Item1 + "" "");
+        System.Console.Write(C1.M().a + "" "");
+        System.Console.Write(C1.M().Item2 + "" "");
+        System.Console.Write(C1.M().b + "" "");
+        System.Console.Write(C2.M().Item1 + "" "");
+        System.Console.Write(C2.M().c + "" "");
+        System.Console.Write(C2.M().Item2 + "" "");
+        System.Console.Write(C2.M().d);
+    }
+}
+";
+            var comp1 = CreateCompilationWithMscorlib(source1);
+            var comp2 = CreateCompilationWithMscorlib(source2);
+            var comp = CompileAndVerify(source, expectedOutput: @"1 1 2 2 3 3 4 4", additionalRefs: new[] { new CSharpCompilationReference(comp1), new CSharpCompilationReference(comp2) });
```

Comment:
It looks like we are missing a test for error scenario like this with the same references:
```
var x = C1.M();
x = C2.M();
```

Please add.